Add filters sidenav

mtruj013 opened the pull request:
## Done

- Added filters sidenav for small screens
- Drive by: fixed filter icon alignment

## QA

- View the site locally in your web browser at: https://ubuntu-com-13458.demos.haus/security/cves?status=
    - Be sure to test on mobile, tablet and desktop screen sizes
- Check that sidenav replaces original filters on small screens and that content is the same as per [design](https://www.figma.com/file/nUJEBpOPTW74RqDRuMQXxB/23.04-U.com---CVEs?type=design&node-id=949-22651&mode=design&t=iMJYHgpN4MAcNAL3-0). Note: **actual filtering functionality covered in separate ticket**

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-8198

webteam-app commented:
Demo starting at https://ubuntu-com-13458.demos.haus

petesfrench commented on templates/security/cve/filters/releases-filter.html:
```diff
@@ -0,0 +1,32 @@
+  </label>
+  <label class="p-checkbox">
+    <input type="checkbox" aria-labelledby="22.10" class="p-checkbox__input" />
+    <span class="p-checkbox__label" id="22.10">22.10 kinetic</span>
```
Is this list complete, it seems like there are some more recent versions missing? ie. 23.04, 23.10

mtruj013 replied:
It is yeah, I added the link to the figma file just now and that's what it's supposed to match

petesfrench commented on templates/security/cve/filters/releases-filter.html:
```diff
@@ -0,0 +1,32 @@
+    <span class="p-checkbox__label" id="14.04">14.04 LTS trusty</span>
+  </label>
+</form>
+<a>Show unmaintained releases</a>
```
Where is this supposed to link to? i assume a new page that isn't built yet

mtruj013 replied:
@juanruitina could you confirm where this, the "What is Ubuntu Priority" and "What do statuses mean?" links are supposed to link to?

petesfrench commented on templates/security/cve/filters/status-filter.html:
```diff
@@ -0,0 +1,27 @@
+<form>
+  <label class="p-checkbox">
+    <input type="checkbox" aria-labelledby="fixed" class="p-checkbox__input" />
```
On the current live page there are 8 status filters but only 6 here, is this accurate?

petesfrench commented on templates/security/cve/_cve-filters-sidenav.html:
```diff
@@ -0,0 +1,64 @@
+            </li>
+            <li class="p-accordion__group" id="releases-filter">
+              <div role="heading" aria-level="2">
+                <button type="button" class="p-accordion__tab u-no-padding--right js-drawer-toggle"  id="tab1" aria-controls="tab1-section" aria-expanded="true">
```
```suggestion
                <button type="button" class="p-accordion__tab u-no-padding--right js-drawer-toggle"  id="tab2" aria-controls="tab2-section" aria-expanded="true">
```

petesfrench replied:
There are two tab 1's at the top here

petesfrench commented on static/sass/_pattern_cve.scss:
```diff
@@ -302,4 +302,20 @@
     }
   }
+
+  .cve-filters {
+    .p-side-navigation {
+      #tab1-section {
+        margin-top: 1rem;
+      }
+
+      #affectedPackages {
+        width: auto;
+      }
+    }
```
```suggestion
  .cve-filters {
    .p-side-navigation {
      #tab1-section {
        margin-top: 1rem;
      }

      #affectedPackages {
        width: auto;
      }
    }
```
We should try and only use classes for targeting styles, as the id's could change in the future. The psuedo class `[:first-child](https://developer.mozilla.org/en-US/docs/Web/CSS/:first-child)` might be useful here

petesfrench commented on templates/security/cve/_cve-filters-sidenav.html:
```diff
@@ -0,0 +1,64 @@
+          Toggle filters
+        </a>
+      </div>
+      <div>
```
I don't think this wrapping `<div>` is needed

juanruitina commented:
The "Toggle filters" button behaves as expected on mobile. Please make sure the filters are also hidden in the medium viewport.

Some feedback, dunno if within scope of this specific issue:

- The "Ubuntu releases" filter header toggles the packages field instead.
- Can we make the filters look on mobile the same as in desktop? Same spacing between filters, no borders/separators (too many right now).

mtruj013 commented on templates/security/cve/_cve-filters-desktop.html:
```diff
@@ -0,0 +1,22 @@
+  <p class="p-text--small-caps">Filters</p>
+</div>
+<!-- package filter -->
+{% include "security/cve/filters/package-filter.html" %}
```
No, the labels for this one are actually 2 separate elements. On large screens it's the regular form label, but on the sidenav it's a button that toggles the accordion and the actual form label in package-filter partial is hidden

mtruj013 commented:
Thanks @petesfrench and @juanruitina ! I addressed most of your comments if you could take a look again? 

Juan let me know if this is what you had in mind for the side nav styling (and if the removal of the separators is a change we can expect to see requested in the future so we can upstream it to Vanilla). Also, I added the hrefs for the links under each filter group but I'm still missing the one for "Show unmaintained releases". What is the functionality supposed to be there?

juanruitina commented:
Looks great, UX+1. Let me get back to you with a design explaining "Show unmaintained releases", please don't be blocked by it, we can tackle it when we do filter functionality.

petesfrench reviewed (approved):
LGTM
